core/scheduler/metrics.go

```diff
@@ -78,7 +88,25 @@ func instrumentDuty(duty core.Duty, defSet core.DutyDefinitionSet) {
 	dutyCounter.WithLabelValues(duty.Type.String()).Add(float64(len(defSet)))
 }
 
-// instrumentValidator sets the validator balance.
-func instrumentValidator(pubkey core.PubKey, totalBal eth2p0.Gwei, status string) {
-	balanceGauge.WithLabelValues(string(pubkey), pubkey.String(), status).Set(float64(totalBal))
+// newMetricSubmitter returns a function that sets validator balance and status metric.
+func newMetricSubmitter() func(pubkey core.PubKey, totalBal eth2p0.Gwei, status string) {
+	var (
+		prevBal    = make(map[core.PubKey]eth2p0.Gwei)
+		prevStatus = make(map[core.PubKey]string)
+	)
+
+	return func(pubkey core.PubKey, totalBal eth2p0.Gwei, status string) {
+		balanceGauge.WithLabelValues(string(pubkey), pubkey.String()).Set(float64(totalBal))
+		statusGauge.WithLabelValues(string(pubkey), pubkey.String(), status).Set(1)
+
+		if prev, ok := prevStatus[pubkey]; ok && prev != status { // Validator status changed
+			statusGauge.WithLabelValues(string(pubkey), pubkey.String(), prev).Set(0)
+		}
+		if prev, ok := prevBal[pubkey]; ok && prev != totalBal { // Validator balance changed
+			balanceGauge.WithLabelValues(string(pubkey), pubkey.String()).Set(0)
```

Comment:
think a bit about what this is doing.... it isn't good.

Reply:
yeah sorry, i was fixing this

Reply:
I also need to fix the grafana panels, doing that